src/test/java/jenkins/branch/PrimaryBranchHealthMetricTest.java

```diff
@@ -77,7 +80,9 @@ public void given__multibranch_with_primary__when__reporting_health__then__prima
             prj.invalidateBuildHealthReports();
             FreeStyleProject master = prj.getItem("master");
             assertThat("We now have the master branch", master, notNullValue());
-            assertThat(prj.getBuildHealthReports(), containsInAnyOrder(master.getBuildHealth()));
+            assertThat(prj.getBuildHealthReports(), containsInAnyOrder(
+                    master.getBuildHealthReports().stream().map(Matchers::is).collect(Collectors.toList())
+            ));
```

Comment:
A `reason` in this `assertThat` would help explain the condition being tested for.

Reply:
we want to check that the folder's health reports are the health reports of the master branch. We do not care about the order of the health reports (though it should be the same as long as they all have different `%` values). We need to wrap each health report in a matcher to stop the generic types being wrong

Reply:
@stephenc 
The request was that you update the `assertThat` call to include a `reason` string.